### PR TITLE
Add --help flag for displaying usage information to users.

### PR DESCRIPTION
**Pull request by ZaahidShaik**
## Changes: 
- This PR, adds change for introducing new `--help` option for users to get the usage information for AIM next-flow pipeline. 

## TESTING :
- In the `AI_MARRVEL` dir, run the command given below. 
````
nextflow run main.nf --help
````

- expected output : 

````
 N E X T F L O W   ~  version 24.04.2

Launching `main.nf` [loquacious_bohr] DSL2 - revision: 5b9ee9fd9c

Usage: 
      
    nextflow run main.nf [All Args] [Options]
        
Example:  

    nextflow run main.nf --ref_dir   path/to/aim_data_dependencies \
                         --input_vcf path/to/sample.vcf.gz \
                         --input_hpo path/to/sample/hpos.txt \
                         --outdir    path/to/sample/Output \ 
                         --run_id [Sample ID] \
                         --ref_ver [hg19/hg38] 

Args:
  --input_vcf               Path to input VCF file.
  --input_hpo               Path to input HPO file.
  --ref_dir                 Path to aim pipeline dependencies directory.
  --outdir                  Output directory.
  --run_id                  Unique identifier for this run. (default: 1)
  --ref_ver                 Reference genome version [hg38 or hg19] (default: hg19)
                  

Options:
  --exome_filter            Enable exonic filter. Addition will filter out variants outside of exonic region  (default: false)
  --help                    Displays the usage information.

Reference Files:
  --ref_loc                 Path to reference location file
  --ref_to_sym              Path to reference to symbol file
  --ref_sorted_sym          Path to reference sorted symbol file
  --ref_exonic_filter_bed   Path to exonic filter BED file

VEP Annotation:
  --vep_dir_cache           Path to VEP cache directory
  --vep_dir_plugins         Path to VEP plugins directory
  --vep_custom_gnomad       Path to custom gnomAD file for VEP
  --vep_custom_clinvar      Path to custom ClinVar file for VEP
  --vep_custom_hgmd         Path to custom HGMD file for VEP

For detailed information about each process, 
please refer to the documentation. https://ai-marrvel.readthedocs.io/en/latest/ 

````
## QA : 

- Happy Case: Tested cmd `nextflow run main.nf --help` got the expected output.
- Fail Case : Tested  cmd `nextflow run main.nf `, got error and workflow was terminated with output given below as expected.

```
 N E X T F L O W   ~  version 24.04.2

Launching `main.nf` [curious_mercator] DSL2 - revision: 5b9ee9fd9c

WARN: Access to undefined parameter `input_vcf` -- Initialise it to a default value eg. `params.input_vcf = some_value`
ERROR ~ A process input channel evaluates to null -- Invalid declaration `path vcf`

 -- Check script 'main.nf' at line: 387 or see '.nextflow.log' file for more details
 
```

### CODE REVIEW

File: docs/source/nf_usage.txt
Comment (hyunhwan-bcm) [inline docs/source/nf_usage.txt:1]: good idea to separate this as a file! 👍

File: nextflow.config [hunk 1]
```diff
@@ -1,8 +1,12 @@
 params {
     run_id = 1
+    help = false
     ref_ver = "hg19"
     bed_filter = ""
     exome_filter = false
+    
+    // Will add exceptions in validations.
+    ref_dir = "NOT_PROVIDED"
```
Comment (hyunhwan-bcm) [inline nextflow.config:9]: it would be better to leave it empty.

**Review by hyunhwan-bcm — APPROVED** [review]
LGTM, do the one single change then we can merge it. Thanks for doing this quickly!

File: nextflow.config (hunk 1 above)
Comment (ZaahidShaik) [reply nextflow.config:9]: I'll make that change in the PR I'll be making for the parameter validation [issue-49](https://github.com/LiuzLab/AI_MARRVEL/issues/49).

**Pull request merged**